lib/ansible/modules/cloud/vmware/vmware_guest_snapshot.py

```diff
@@ -270,6 +270,7 @@ def snapshot_vm(self, vm):
             self.module.fail_json(msg="Failed to take snapshot due to VMware Licence: %s" % to_native(exc.msg))
         except Exception as exc:
             self.module.fail_json(msg="Failed to create snapshot of VM %s due to %s" % (self.module.params['name'], to_native(exc)))
+        self.wait_for_task(task)
```

Comment:
We wait for task at [line 341](https://github.com/ansible/ansible/blob/135d34325495b38cb91ec03ba55b3b1c56c06b72/lib/ansible/modules/cloud/vmware/vmware_guest_snapshot.py#L341)